Dockerfile.s3

```diff
@@ -0,0 +1,154 @@
+# base: Stage which installs necessary runtime dependencies (OS packages, java,...)
+FROM python:3.11.5-slim-buster@sha256:9f35f3a6420693c209c11bba63dcf103d88e47ebe0b205336b5168c122967edf as base
+ARG TARGETARCH
+
+# set workdir
+RUN mkdir -p /opt/code/localstack
+WORKDIR /opt/code/localstack/
+
+# Install runtime OS package dependencies
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt-get update && \
+        # Install dependencies to add additional repos
+        apt-get install -y --no-install-recommends \
+            # Runtime packages (groff-base is necessary for AWS CLI help)
+             ca-certificates curl make openssl
+# do we need the dns server?
+#             ca-certificates curl make openssl iputils-ping iproute2
+# default under
+#             ca-certificates curl gnupg make openssl tar pixz zip unzip groff-base iputils-ping nss-passwords procps iproute2 xz-utils
+
+SHELL [ "/bin/bash", "-c" ]
+
+# create localstack user and filesystem hierarchy, perform some permission fixes
+RUN chmod 777 . && \
+    useradd -ms /bin/bash localstack && \
+    mkdir -p /var/lib/localstack && \
+    chmod -R 777 /var/lib/localstack && \
+    mkdir -p /usr/lib/localstack && \
+    mkdir /tmp/localstack && \
+    chmod -R 777 /tmp/localstack && \
+    touch /tmp/localstack/.marker
+
+# install basic (global) tools to final image
+RUN --mount=type=cache,target=/root/.cache \
+    pip install --no-cache-dir --upgrade virtualenv
+
+# install the entrypoint script
+ADD bin/docker-entrypoint.sh /usr/local/bin/
+# add the shipped hosts file to prevent performance degredation in windows container mode on windows
+# (where hosts file is not mounted) See https://github.com/localstack/localstack/issues/5178
+ADD bin/hosts /etc/hosts
+
+# expose default environment
+# Set edge bind host so localstack can be reached by other containers
+# set library path and default LocalStack hostname
+ENV USER=localstack
+ENV PYTHONUNBUFFERED=1
+
+# Install the latest version of awslocal globally
+# RUN --mount=type=cache,target=/root/.cache \
+#     pip3 install --upgrade awscli awscli-local requests
```

Comment:
I commented out the globally installed awscli inside the container, as it's really heavy. Now that they compress the specs, maybe we could add it but maybe it could be opt-it?